### PR TITLE
Correct simple invoice amount limits

### PR DESCRIPTION
**Pull request by thenmaster**

### CODE REVIEW

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/builders/impl/PTSimpleInvoiceBuilderImpl.java [hunk 1]
```diff
@@ -74,11 +72,11 @@ protected void validateInstance() throws BillyValidationException {
                 PTGenericInvoiceBuilderImpl.LOCALIZER.getString("field.clientType"));
         super.validateInstance();
 
-        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal(1000)) >= 0) {
-            throw new BillySimpleInvoiceException("Amount > 1000 for customer simple invoice. Issue invoice");
+        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal("3000")) >= 0) {
+            throw new InvalidAmountForDocumentTypeException("Amount > 3000 for customer simple invoice. Issue invoice");
```
Comment (Vasco-F) [inline billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/builders/impl/PTSimpleInvoiceBuilderImpl.java:76]: Isn't the value 1000€ for end customers for simplified invoices in Portugal?

File: billy-spain/src/main/java/com/premiumminds/billy/spain/services/builders/impl/ESSimpleInvoiceBuilderImpl.java [hunk 2]
```diff
@@ -65,11 +64,11 @@ protected void validateInstance() throws BillyValidationException {
                 ESGenericInvoiceBuilderImpl.LOCALIZER.getString("field.clientType"));
         super.validateInstance();
 
-        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal(1000)) >= 0) {
-            throw new BillySimpleInvoiceException("Amount > 1000 for customer simple invoice. Issue invoice");
+        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal("1000")) >= 0) {
+            throw new InvalidAmountForDocumentTypeException("Amount > 1000 for customer simple invoice. Issue invoice");
```
Comment (Vasco-F) [inline billy-spain/src/main/java/com/premiumminds/billy/spain/services/builders/impl/ESSimpleInvoiceBuilderImpl.java:68]: Shouldn't this value be 3000€ for ClientType.Customer?

**Review by Vasco-F — CHANGES_REQUESTED** [review]

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/builders/impl/PTSimpleInvoiceBuilderImpl.java (hunk 1 above)
Comment (thenmaster) [reply billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/builders/impl/PTSimpleInvoiceBuilderImpl.java:76]: Indeed it is. I misread my notes and mixed up the values for PT and ES.

File: billy-spain/src/main/java/com/premiumminds/billy/spain/services/builders/impl/ESSimpleInvoiceBuilderImpl.java (hunk 2 above)
Comment (thenmaster) [reply billy-spain/src/main/java/com/premiumminds/billy/spain/services/builders/impl/ESSimpleInvoiceBuilderImpl.java:68]: Indeed it is. I misread my notes and mixed up the values for PT and ES.

**Review by Vasco-F — APPROVED** [review]

File: billy-andorra/src/main/java/com/premiumminds/billy/andorra/services/builders/impl/ADSimpleInvoiceBuilderImpl.java [hunk 3]
```diff
@@ -69,11 +68,11 @@ protected void validateInstance() throws BillyValidationException {
                                  ADGenericInvoiceBuilderImpl.LOCALIZER.getString("field.clientType"));
         super.validateInstance();
 
-        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal(1000)) >= 0) {
-            throw new BillySimpleInvoiceException("Amount > 1000 for customer simple invoice. Issue invoice");
+        if (i.getClientType() == CLIENTTYPE.CUSTOMER && i.getAmountWithTax().compareTo(new BigDecimal("40000")) >= 0) {
```
Comment (froque) [inline billy-andorra/src/main/java/com/premiumminds/billy/andorra/services/builders/impl/ADSimpleInvoiceBuilderImpl.java:71]: shoud these hardcoded values be moved to configuration options ?

File: billy-andorra/src/main/java/com/premiumminds/billy/andorra/exceptions/BillySimpleInvoiceException.java
Comment (froque) [inline billy-andorra/src/main/java/com/premiumminds/billy/andorra/exceptions/BillySimpleInvoiceException.java:1]: maybe do not delete these files, but mark them as `@Deprecated(forRemoval=true)` and we remove them on the next major version ?

File: billy-andorra/src/main/java/com/premiumminds/billy/andorra/services/builders/impl/ADSimpleInvoiceBuilderImpl.java (hunk 3 above)
Comment (thenmaster) [reply billy-andorra/src/main/java/com/premiumminds/billy/andorra/services/builders/impl/ADSimpleInvoiceBuilderImpl.java:71]: That's a good idea :+1:

**Review by froque — APPROVED** [review]

**Pull request merged**